packages/arb-token-bridge-ui/src/token-bridge-sdk/utils.ts

```diff
@@ -58,3 +69,55 @@ export function percentIncrease(
 ): BigNumber {
   return num.add(num.mul(increase).div(100))
 }
+
+// We cannot hardcode Erc20Bridger anymore in code, especially while dealing with tokens
+// this function returns the Bridger matching the set providers
+export const getBridger = async ({
+  sourceChainProvider,
+  destinationChainProvider,
+  isNativeCurrencyTransfer = false
+}: {
+  sourceChainProvider: Provider
+  destinationChainProvider: Provider
+  isNativeCurrencyTransfer?: boolean
+}) => {
+  const sourceChainId = await getChainIdFromProvider(sourceChainProvider)
```

Comment:
looks like you can simply pass `sourceChain.id` into this function rather than deriving it from the provider

Reply:
Updated